.github/workflows/release.yml

```diff
@@ -0,0 +1,53 @@
+---
+name: Build and Optionally Release
+
+on:  # yamllint disable-line rule:truthy
+  pull_request:
+  push:
+    branches:
+    - main
```

Comment:
I'm wondering if we wouldn't want to separate the "build on every commit" from the actual image publish stage (on ideally hook that onto tag/[release](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release) creation ? WDYT?

Reply:
I have split up the unique workflow into two different workflows (one to build on PRs and push to main, and one to push on release).